website/blog/2019-01-00-1.16.0.md

```diff
@@ -0,0 +1,370 @@
+
+There's a `--list-different` flag that is used to fail the CI if one of the
+input files is not formatted, which works great but we found that its _error
+message_ is not human-friendly but machine-friendly, so we added a `--check`
+flag to output something human-friendly.
+
+<code style="display: block; background: #282a36; color: #f1f1f0">
+<span style="color: #686868"># Prettier 1.15</span><br />
+$ <span style="color: #5af78e">prettier</span> <u>*.js</u> --list-different<br />
+unformatted.js<br />
+<br />
+<span style="color: #686868"># Prettier 1.16</span><br />
+$ <span style="color: #5af78e">prettier</span> <u>*.js</u> --check<br />
+Checking formatting...<br />
+unformatted.js<br />
+Code style issues found in the above file(s). Forgot to run Prettier?<br />
+</code>
+
+## Other changes
+
+### General
+
+#### Fix unexpected formatting caused by range format ([#5632] by [@lhchavez])
+
+<!-- prettier-ignore -->
+```js
+// Input
+something("something", () => {
+  const something = {
+    something: [
+      {
+        long1: "longlonglonglonglonglonglonglonglonglonglonglong",
+        long2: "longlonglonglonglonglonglonglonglonglonglonglong",
+      }
+    ]
+  };
+});
+
+// Output (Prettier 1.15, --range-start 100 --range-end 120)
+something("something", () => {
+  const something = { something: [{ long1: "longlonglonglonglonglonglonglonglonglonglonglong", long2: "longlonglonglonglonglonglonglonglonglonglonglong" }] };
+});
+
+// Output (Prettier 1.16, --range-start 100 --range-end 120)
+something("something", () => {
+  const something = {
+    something: [
+      {
+        long1: "longlonglonglonglonglonglonglonglonglonglonglong",
+        long2: "longlonglonglonglonglonglonglonglonglonglonglong",
+      }
+    ]
+  };
+});
+```
+
+### Standalone
+
+#### Remove the dynamic `require()` call in the standalone bundle ([#5612] by [@j-f1])
+
+Previously, dynamic `require()`s are still presented in the standalone bundle
+even if it's not used, which causes a warning if you'd like to bundle it. We
+have removed those unnecessary `require()`s in 1.16.
+
+### TypeScript
+
+#### Correctly handle `//` in TSX ([#5728] by [@JamesHenry])
+
+Previously, `//`s in TSX were wrongly treated as comments, it's now fixed.
+
+<!-- prettier-ignore -->
+```js
+// Input
+const link = <a href="example.com">http://example.com</a>
+
+// Output (Prettier 1.15)
+// Error: Comment location overlaps with node location
+
+// Output (Prettier 1.16)
+const link = <a href="example.com">http://example.com</a>;
+```
+
+#### Remove redundant parentheses around type annotations ([#5724] by [@flurmbo])
+
+<!-- prettier-ignore -->
+```js
+// Input
+class Foo {
+  bar: (() => boolean);
+}
+
+// Output (Prettier 1.15)
+class Foo {
+  bar: (() => boolean);
+}
+
+// Output (Prettier 1.16)
+class Foo {
+  bar: () => boolean;
+}
+```
+
+### JavaScript
+
+#### TODO(title): fix(javascript): skip .connect() method when composing fun ([#5739] by [@makepost])
+
+TODO
```

Comment:
Not quite sure what's the reason behind it, why do we need to add a special special case for `connect`?

cc @j-f1 @suchipi

Reply:
The reason is that even though Redux and friends are very common, there are lots of other just as common and valid cases, and they can be summed up as `anything_that_might_need_a_connection.connect()` (sockets, DBs, etc.)